src/vs/code/electron-main/app.ts

```diff
@@ -359,6 +359,11 @@ export class CodeApplication extends Disposable {
 		// We handle uncaught exceptions here to prevent electron from opening a dialog to the user
 		setUnexpectedErrorHandler(error => this.onUnexpectedError(error));
 		process.on('uncaughtException', error => {
+			// Don't process the exception if exit is happening as it can cause the main process to
+			// hang while awaiting an IPC reply.
+			if (this.lifecycleMainService.quitRequested) {
```

Comment:
Do we send telemetry during shutdown ? Wondering if the check should be moved before https://github.com/microsoft/vscode/blob/8e386a6a9607d541a37f34a11a98e19c7f1f5829/src/vs/base/common/errors.ts#L24 so that we don't rethrow on shutdown but still end up sending telemetry ?